src/daft-recordbatch/src/ops/window_states/count.rs

```diff
@@ -0,0 +1,54 @@
+use arrow2::bitmap::Bitmap;
+use common_error::DaftResult;
+use daft_core::prelude::*;
+
+use super::WindowAggStateOps;
+
+pub struct CountWindowState {
+    source: Bitmap,
+    count: usize,
+    count_vec: Vec<u64>,
+}
+
+impl CountWindowState {
+    pub fn new(source: &Series, total_length: usize) -> Self {
+        let source_bitmap = source
+            .validity()
+            .cloned()
+            .unwrap_or_else(|| Bitmap::new_constant(true, source.len()));
```

Comment:
Instead of creating this true bitmap here, could we just store the `Option<Bitmap>` here? Then, just have different implementations for `add` and `remove`.

Also, could we just store the reference to the bitmap here instead of cloning? I believe it would need to be an `Option<&'a Bitmap>`, and the struct would need to have a specified lifetime as well (`CountWindowState<'a>`). (If not, don't worry too much since cloning a bitmap doesn't look that expensive)

Reply:
Another note actually: There are multiple different count modes, which handle validity differently. The count mode can be obtained from the second argument to `AggExpr::Count`. Let's handle those correctly here too